src/components/TablePanel.vue

```diff
@@ -59,93 +68,137 @@
+              <v-btn icon>
+                <v-icon
+                  color="red accent-3"
+                  @click="deleteItem(item)"
+                >delete</v-icon>
               </v-btn>
             </v-list-item-action>
           </v-list-item>
         </v-hover>
       </v-list-item-group>
     </template>
     <div
-      class="ws-detail-empty-list"
       v-else
+      class="ws-detail-empty-list"
     >
       <v-icon color="blue lighten-1">info</v-icon> There's nothing here yet...
     </div>
   </v-list>
 </template>
 
 <script lang="ts">
-import Vue from 'vue';
+import Vue, { PropType } from 'vue';
+import DeleteTableDialog from '@/components/DeleteTableDialog.vue';
 import TableDialog from '@/components/TableDialog.vue';
-import GraphDialog from '@/components/GraphDialog.vue';
+import DownloadDialog from '@/components/DownloadDialog.vue';
 
 export default Vue.extend({
-  name: 'ItemPanel',
+  name: 'TablePanel',
+
   components: {
-    GraphDialog,
+    DeleteTableDialog,
     TableDialog,
+    DownloadDialog,
   },
+
   props: {
-    title: {
-      type: String,
-      required: true,
-    },
-    items: {
-      type: Array,
-      required: true,
-    },
     workspace: {
-      type: String,
-      required: true,
-    },
-    routeType: {
-      type: String,
+      type: String as PropType<string>,
       required: true,
     },
-    icon: {
-      type: String,
+
+    items: {
+      type: Array as PropType<string[]>,
       required: true,
     },
   },
+
   data() {
     interface CheckboxTable {
       [index: string]: boolean;
     }
 
     return {
       checkbox: {} as CheckboxTable,
+      singleSelected: null as string | null,
+      deleterDialog: false,
+      downloaderDialog: false,
     };
   },
+
   computed: {
-    selection(): string[] {
+    checked(): string[] {
       return Object.keys(this.checkbox)
         .filter((d) => !!this.checkbox[d]);
     },
+
+    selection(): string[] {
+      const {
+        singleSelected,
+        checked,
+      } = this;
+
+      let result: string[] = [];
+
+      if (singleSelected !== null) {
+        result = result.concat([singleSelected]);
+      } else {
+        result = result.concat(checked);
+      }
+
+      return result;
+    },
+
     anySelected(): boolean {
       return this.selection.length > 0;
     },
   },
 
   methods: {
+    deleteItem(item: string) {
+      this.singleSelected = item;
+      (this.$refs.deleter as any).dialog = true;
+    },
+
+    downloadItem(item: string) {
+      this.singleSelected = item;
+      (this.$refs.downloader as any).dialog = true;
+    },
+
     clearCheckboxes() {
       Object.keys(this.checkbox).forEach((key) => {
         this.checkbox[key] = false;
       });
     },
+
+    cleanup(deleted?: string[]) {
+      const {
+        singleSelected,
+      } = this;
+
+      if (singleSelected !== null) {
+        this.singleSelected = null;
+      }
```

Comment:
Can you simplify this to just: `this.singleSelected = null`?

Reply:
See 14dcbf1dbaf16da19aa4551782d5a6419fad6314.

Reply:
Not a huge deal, but I think you can also delete the 
```
     const {
        singleSelected,
      } = this;
```

Reply:
Good catch. See f25502a.